tawk_to.module

```diff
@@ -1,316 +1,200 @@
+    return;
+  }
 
-/**
- * Returns help text which is dispalyed in modules page.
- */
-function tawk_to_help($path, $arg)
-{
-    switch ($path) {
-        case "admin/help#tawk_to":
-            return '<p>' . t("Tawk.to widget cutomization, using this module you can select widget which will be used on every page on your site") . '</p>';
-        break;
-    }
-}
+  $widget = TawkHelper::get_widget();
 
-/**
- * Retrieves widget details from database.
- */
-function tawk_to_get_widget()
-{
-    return array(
-            'page_id' => variable_get(TAWKTO_WIDGET_PAGE_ID, ''),
-            'widget_id' => variable_get(TAWKTO_WIDGET_WIDGET_ID, ''),
-        );
-}
+  if ($widget['page_id'] === '' || $widget['widget_id'] === '') {
+    return null;
+  }
 
-/**
- * Hook for adding widget script to every page.
- */
-function tawk_to_page_build(&$page)
-{
-    global $base_url;
+  $page_id = $widget['page_id'];
+  $widget_id = $widget['widget_id'];
 
-    // backend checking; do not show chat widget on admin page(s)
-    if (path_is_admin(current_path())) {
-        return;
-    }
+  // get visibility options
+  $options = variable_get(TawkHelper::TAWK_TO_WIDGET_OPTIONS, '');
+
+  // initial markup options
+  $enable_visitor_recognition = true; // default value
 
-    $widget = tawk_to_get_widget();
+  if ($options) {
+    $options = json_decode($options);
 
-    if ($widget['page_id'] === '' || $widget['widget_id'] === '') {
-        return null;
+    if (isset($options->enable_visitor_recognition)) {
+      $enable_visitor_recognition = $options->enable_visitor_recognition;
     }
-    $page_id = $widget['page_id'];
-    $widget_id = $widget['widget_id'];
 
-    // get visibility options
-    $options = variable_get(TAWKTO_WIDGET_OPTIONS, '');
-    $markupOptions = array(
-        'enable_visitor_recognition' => true, // default value
-    );
-    if ($options) {
-        $options = json_decode($options);
+    // prepare visibility
+    $currentUrl = $base_url.$_SERVER['REQUEST_URI'];
+
+    if (false === $options->always_display) {
+      $showPages = json_decode($options->show_oncustom);
+      $show = false;
+
+      foreach ($showPages as $slug) {
+        if ($currentUrl === $slug) {
+          $show = true;
+          break;
+        }
+      }
 
-        if (isset($options->enable_visitor_recognition)) {
-            $markupOptions['enable_visitor_recognition'] = $options->enable_visitor_recognition;
+      if (arg(0) === 'taxonomy' && arg(1) === 'term' && is_numeric(arg(2)) && empty(arg(3))) {
+        if (true === $options->show_oncategory) {
+          $show = true;
         }
+      }
+
+      // front page
+      if (drupal_is_front_page()) {
+        if (true === $options->show_onfrontpage) {
+          $show = true;
+        }
+      }
+
+      if (!$show) {
+        return;
+      }
+    } else {
+      if (isset($options->hide_oncustom)) {
+        $hide_pages = json_decode($options->hide_oncustom);
```

Comment:
Inconsistent casing in variable names (``$showPages` vs `$hide_pages`)

Reply:
Done!

Reply:
Updated $showPages and $currentUrl to $show_pages and $current_url since most of the variable declarations are already in snake case